Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputMapper.cpp

```diff
@@ -408,32 +439,76 @@ namespace AzToolsFramework
 
     void QtEventToAzInputMapper::HandleKeyEvent(QKeyEvent* keyEvent)
     {
-        // Ignore key repeat events, they're unrelated to actual physical button presses.
+        const Qt::Key key = static_cast<Qt::Key>(keyEvent->key());
+        const QEvent::Type eventType = keyEvent->type();
+
+        // special handling for text events in edit mode
+        {
+            static Qt::Key lastKey = key;
+            static bool consumeAsText = false;
+
+            QString keyText = keyEvent->text();
+            if (key == Qt::Key_Backspace)
+            {
+                keyText = "\b";
+            }
+
+            if (!keyText.isEmpty())
+            {
+                // key events are first sent as shortcuts, if accepted they are then resent as traditional key
+                // down events.  dispatching the key event as text on shortcut (and auto-repeat press) ensures
```

Comment:
question: is `text on shortcut` correct? I'm not sure I understand that part

Reply:
Ah okay, potentially we could update it to read...

```
dispatching the key event as text during a shortcut
```

or

```
dispatching the key event as text on the shortcut event
```

Also we should update `resent` to `re-sent` (resent is to feel bitterness or indignation 😛)